phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCStreamIT.java

```diff
@@ -433,6 +434,34 @@ public void testStreamMetadataWhenTableIsDropped() throws SQLException {
     Assert.assertTrue(rs.next());
   }
 
+    @Test
+    public void testStreamMetadataWhenCDCIsDropped() throws SQLException {
+        Connection conn = newConnection();
+        MetaDataClient mdc = new MetaDataClient(conn.unwrap(PhoenixConnection.class));
+        String schemaName = "\"" + generateUniqueName().toLowerCase() + "\"";
+        String tableName =
+                SchemaUtil.getTableName(schemaName, "\"" + generateUniqueName().toLowerCase() + "\"");
+        String unescapedFullTableName = SchemaUtil.getUnEscapedFullName(tableName);
+        String create_table_sql =
+                "CREATE TABLE  " + tableName + " ( k INTEGER PRIMARY KEY," + " v1 INTEGER, v2 DATE)";
+        conn.createStatement().execute(create_table_sql);
+        String cdcName = "\"" + generateUniqueName().toLowerCase() + "\"";
+        String cdc_sql = "CREATE CDC " + cdcName + " ON " + tableName;
+        conn.createStatement().execute(cdc_sql);
+        TaskRegionObserver.SelfHealingTask task = new TaskRegionObserver.SelfHealingTask(
+                taskRegionEnvironment, QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS);
+        task.run();
+        String drop_cdc_sql = "DROP CDC " + cdcName + " ON " + tableName;
```

Comment:
nit: from Java variable convention, we can use `dropCdcSql` but it's fine for now, we can do this cleanup later also